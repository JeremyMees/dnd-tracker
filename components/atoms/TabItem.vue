<script setup lang="ts">
const props = withDefaults(
  defineProps<{
    link: string
    icon: string
    label: string
    disabled?: boolean
  }>(), {
    disabled: false,
  },
)

const route = useRoute()

const isActive = computed<boolean>(() => route.fullPath.includes(props.link))
</script>

<template>
  <div
    class="p-1 rounded-lg lg:rounded-none lg:bg-transparent lg:px-0 lg:py-0 lg:border-b-2 lg:relative lg:top-[2px] transition-colors duration-200 ease-in-out"
    :class="[
      disabled && 'pointer-events-none opacity-50',
      isActive
        ? 'bg-primary/50 lg:border-primary'
        : 'bg-secondary/50 lg:border-transparent',
    ]"
  >
    <NuxtLinkLocale
      :to="link"
      class="text-sm flex gap-2 items-center py-1 px-2 rounded-lg text-foreground hover:bg-tracker hover:foreground"
      active-class="text-foreground"
    >
      <Icon
        :name="icon"
        aria-hidden="true"
        class="size-4"
      />
      <span>
        {{ label }}
      </span>
    </NuxtLinkLocale>
  </div>
</template>
