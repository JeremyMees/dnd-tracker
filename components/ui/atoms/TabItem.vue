<script setup lang="ts">
const props = withDefaults(
  defineProps<{
    link: string
    icon: string
    label: string
    disabled?: boolean
  }>(), {
    disabled: false,
  },
)

const route = useRoute()

const isActive = computed<boolean>(() => route.fullPath.includes(props.link))
</script>

<template>
  <div
    class="p-1 rounded-lg md:rounded-none md:bg-transparent md:px-0 md:py-0 md:border-b-2 md:relative md:top-[2px] transition-colors duration-200 ease-in-out"
    :class="[
      disabled && 'pointer-events-none opacity-50',
      isActive
        ? 'bg-primary/50 md:border-primary'
        : 'bg-slate-700/50 md:border-transparent',
    ]"
  >
    <RouteLink
      :url="link"
      class="flex gap-2 items-center py-1 px-2 rounded-lg text-slate-300 hover:bg-tracker hover:text-white"
      active-class="text-white"
    >
      <Icon
        :name="icon"
        aria-hidden="true"
        class="size-5"
      />
      <span>
        {{ label }}
      </span>
    </RouteLink>
  </div>
</template>
