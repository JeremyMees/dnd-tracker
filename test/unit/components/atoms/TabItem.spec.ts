import { mockNuxtImport, mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'
import TabItem from '~/components/atoms/TabItem.vue'

interface Props {
  link: string
  icon: string
  label: string
  disabled: boolean
}

const props: Props = {
  link: '/home',
  icon: 'home',
  label: 'Home',
  disabled: false,
}

describe('TabItem', async () => {
  mockNuxtImport('useRoute', () => () => ({
    fullPath: '/home',
  }))

  it('Should match snapshot', async () => {
    const component = await mountSuspended(TabItem, { props })

    expect(component.html()).toMatchSnapshot()
  })

  it('Should show inactive state', async () => {
    const component = await mountSuspended(TabItem, { props: { ...props, link: '/other-path' } })

    expect(component.attributes('class')).toContain('bg-secondary/50 lg:border-transparent')
  })

  it('Should show active state', async () => {
    const component = await mountSuspended(TabItem, { props })

    expect(component.attributes('class')).toContain('bg-primary/50 lg:border-primary')
  })

  it('Should show disabled state', async () => {
    const component = await mountSuspended(TabItem, { props: { ...props, disabled: true }, global })

    expect(component.attributes('class')).toContain('pointer-events-none opacity-50')
  })
})
