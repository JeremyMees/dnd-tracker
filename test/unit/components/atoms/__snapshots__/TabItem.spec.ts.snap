// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TabItem > Should match snapshot 1`] = `"<div class="p-1 rounded-lg lg:rounded-none lg:bg-transparent lg:px-0 lg:py-0 lg:border-b-2 lg:relative lg:top-[2px] transition-colors duration-200 ease-in-out bg-primary/50 lg:border-primary"><a class="text-sm flex gap-2 items-center py-1 px-2 rounded-lg text-foreground hover:bg-tracker hover:foreground"><span class="iconify i-home size-4" aria-hidden="true"></span><span>Home</span></a></div>"`;
